### PR TITLE
refactor: don't call `get_available_cameras` each time camera is switched on camera console, slightly optimize `get_available_cameras`

### PR DESCRIPTION
**Pull request by Gaxeer**
## What Does This PR Do
Optimize camera console slightly (`get_available_cameras` proc)
Don't call `get_available_cameras` each time camera is switched on camera console.

## Why It's Good For The Game
Less lags, cleaner code.

## Testing
Compiles, cameras works. Didn't bench it.

### CODE REVIEW

File: code/modules/mob/living/silicon/ai/freelook/cameranet.dm [hunk 1]
```diff
@@ -8,13 +8,32 @@ GLOBAL_DATUM_INIT(cameranet, /datum/cameranet, new())
...
 	// The chunks of the map, mapping the areas that the cameras can see.
 	var/list/chunks = list()
 	var/ready = FALSE
 
 	// The object used for the clickable stat() button.
 	var/obj/effect/statclick/statclick
 
+/datum/cameranet/proc/register_camera(obj/machinery/camera/camera)
+	if(!istype(camera))
+		CRASH("Not `/obj/machinery/camera` was tried to be registered")
```
Comment (DGamerL) [inline code/modules/mob/living/silicon/ai/freelook/cameranet.dm:22]: Could you make it tell you what instead was being attempted to registered?

File: code/modules/mob/living/silicon/ai/freelook/cameranet.dm [hunk 2]
```diff
@@ -8,13 +8,32 @@ GLOBAL_DATUM_INIT(cameranet, /datum/cameranet, new())
...
+
+	cameras |= camera
+	cameras_by_tag[camera.c_tag] = camera
+
+/datum/cameranet/proc/unregister_camera(obj/machinery/camera/camera)
+	if(!istype(camera))
+		CRASH("Not `/obj/machinery/camera` was tried to be unregistered")
```
Comment (DGamerL) [inline code/modules/mob/living/silicon/ai/freelook/cameranet.dm:29]: Same here

File: code/modules/mob/living/silicon/ai/freelook/cameranet.dm [hunk 3]
```diff
@@ -178,21 +197,3 @@ GLOBAL_DATUM_INIT(cameranet, /datum/cameranet, new())
 		if(chunk.visibleTurfs[position])
 			return 1
 	return 0
```
Comment (DGamerL) [inline code/modules/mob/living/silicon/ai/freelook/cameranet.dm:199]:
```suggestion
			return TRUE
	return FALSE
```
While we're here

File: code/game/machinery/computer/camera_console.dm [hunk 4]
```diff
@@ -147,23 +149,30 @@
...
-		if(tempnetwork.len)
-			D["[C.c_tag]"] = C
-	return D
+
+		cameras_by_tag["[camera.c_tag]"] = camera
+
+	return cameras_by_tag
+
+/obj/machinery/computer/security/proc/is_camera_available(obj/machinery/camera/camera_to_check, console_on_away_level)
+	if((camera_to_check.z != z) && (console_on_away_level || is_away_level(camera_to_check.z)))
```
Comment (lewcc) [inline code/game/machinery/computer/camera_console.dm:163]:
```suggestion
	if(camera_to_check.z != z && (console_on_away_level || is_away_level(camera_to_check.z)))
```

File: code/game/machinery/computer/camera_console.dm [hunk 5]
```diff
@@ -147,23 +149,30 @@
...
+		return FALSE
+
+	var/camera_network = camera_to_check.network
+	if(!camera_network)
+		stack_trace("Camera in a cameranet has no camera network")
+		return FALSE
+
+	if(!(islist(camera_network)))
```
Comment (lewcc) [inline code/game/machinery/computer/camera_console.dm:171]:
```suggestion
	if(!islist(camera_network))
```

**Pull request closed without merging**